delocate/tools.py

```diff
@@ -246,18 +433,27 @@ def get_install_names(filename):
     install_names : tuple
         tuple of install names for library `filename`
     """
-    lines = _cmd_out_err(["otool", "-L", filename])
-    if not _line0_says_object(lines[0], filename):
+    otool = subprocess.run(
+        ["otool", "-L", filename],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if not _line0_says_object(otool.stdout or otool.stderr, filename):
         return ()
-    names = tuple(parse_install_name(line)[0] for line in lines[1:])
     install_id = get_install_id(filename)
-    if install_id is not None:
-        assert names[0] == install_id
-        return names[1:]
-    return names
-
-
-def get_install_id(filename):
+    names_data = _ignore_architectures(_parse_otool_install_names(otool.stdout))
```

Comment:
Maybe comment here about asserting values identical for multiple archs?

Reply:
This is now indicated on the functions `Raises` section.